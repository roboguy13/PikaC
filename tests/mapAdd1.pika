%generate mapAdd1

Sll : layout[x](List);
Sll Nil := emp;
Sll (Cons head tail) :=
  x :-> head **
  (x+1) :-> nxt **
  Sll tail [nxt];

mapAdd1 : Sll -> Sll;
mapAdd1 Nil := Nil [Sll];
mapAdd1 (Cons h t) :=
  Cons (h + 1) (mapAdd1 t) [Sll];

