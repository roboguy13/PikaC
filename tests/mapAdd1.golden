*** Running test mapAdd1
(2 (3 (4 (5 ()))))

